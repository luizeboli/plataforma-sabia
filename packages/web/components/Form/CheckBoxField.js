import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledCheckBox = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	padding: 1rem;
	font: 1.2rem;
`;

const StyledCheckBoxInput = styled.input`
	width: 2.5rem;
	height: 2.5rem;
	margin: 0rem;
	margin: 0 0.5rem 0 0;
`;
const StyledCheckBoxLabel = styled.div`
	width: 100%;
	color: ${({ theme }) => theme.colors.lightGray};
`;

const CheckBoxField = ({ name, value, label, required, onChange }) => {
	const handleOnChange = () => onChange((oldValue) => !oldValue);
	return (
		<StyledCheckBox>
			<StyledCheckBoxInput
				id={name}
				name={name}
				type="checkbox"
				aria-label={label}
				aria-required={required}
				required={required}
				checked={value}
				onChange={handleOnChange}
			/>
			<StyledCheckBoxLabel>{label}</StyledCheckBoxLabel>
		</StyledCheckBox>
	);
};
CheckBoxField.propTypes = {
	name: PropTypes.string.isRequired,
	value: PropTypes.bool,
	label: PropTypes.string,
	required: PropTypes.bool,
	onChange: PropTypes.func,
};

CheckBoxField.defaultProps = {
	value: false,
	label: '',
	required: false,
	onChange: () => {},
};

export default CheckBoxField;
