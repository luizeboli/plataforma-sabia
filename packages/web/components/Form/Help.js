import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { AiFillQuestionCircle } from 'react-icons/ai';
import ReactTooltip from 'react-tooltip';

const Icon = styled(AiFillQuestionCircle)`
	width: 4rem;
	height: 4rem;
	margin: 0.6rem 0.5rem;
	fill: ${({ theme }) => theme.colors.mediumGray};
	transition: fill 0.2s;
	:hover {
		cursor: pointer;
		fill: ${({ theme }) => theme.colors.darkGray};
	}
`;

const Help = ({ id, HelpComponent }) => {
	return (
		<>
			<Icon data-tip data-for={id} />
			<ReactTooltip uuid="uuid" id={id} multiline type="dark" place="bottom" effect="solid">
				{HelpComponent}
			</ReactTooltip>
		</>
	);
};

Help.propTypes = {
	id: PropTypes.string,
	HelpComponent: PropTypes.node,
};

Help.defaultProps = {
	id: '',
	HelpComponent: <p>Help Text</p>,
};

export default Help;
