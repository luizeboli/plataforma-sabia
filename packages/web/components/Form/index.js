export { default as InputField } from './InputField';
export { default as SwitchField } from './SwitchField';
export { default as TextField } from './TextField';
export { default as SelectField } from './SelectField';
export { default as CheckBoxField } from './CheckBoxField';
export { default as MaskedInputField } from './MaskedInputField';
export { default as Watcher } from './Watcher';
export { default as InputHiddenField } from './InputHiddenField';
export { default as Repeater } from './Repeater';
export { default as Help } from './Help';
export * from './Form';
