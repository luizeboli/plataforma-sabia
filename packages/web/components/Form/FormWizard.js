import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { AiOutlineCheck } from 'react-icons/ai';
import { Form, Actions } from './Form';
import { Button } from '../Button';

const FormWizardContainer = styled.div``;

const StepsContainer = styled.div`
	width: 100%;
	background: ${({ theme }) => theme.colors.primary};
	border-top: 4px solid ${({ theme }) => theme.colors.lightGray};
	padding: 3rem 0 5rem 0;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		padding: 1rem 0 4rem 0;
	}
`;

const MobileSteps = styled.ol`
	display: flex;
	flex-direction: column;
	align-items: center;

	div {
		display: flex;
		justify-content: center;
		align-items: center;
		list-style: none;

		> li:nth-child(2) {
			font-size: 6rem;
		}

		> li:not(:last-child) {
			margin-right: 2rem;
		}
	}

	@media (min-width: ${({ theme }) => theme.screens.medium}px) {
		display: none;
	}
`;

const WebSteps = styled.ol`
	display: flex;
	justify-content: center;
	align-items: center;
	list-style: none;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		display: none;
	}
`;

const StepItem = styled.li`
	position: relative;
	display: flex;
	align-items: center;

	${({ completed }) =>
		completed &&
		css`
			> div {
				> p {
					color: ${({ theme }) => theme.colors.darkGray};
				}
				> span {
					color: ${({ theme }) => theme.colors.white};
					background: ${({ theme }) => theme.colors.darkGray};
				}
			}
		`};

	> div {
		display: flex;
		align-items: center;
		flex-direction: column;

		> p {
			height: 2rem;
		}
	}

	::after {
		content: '';
		display: block;
		position: relative;
		width: 100px;
		height: 12px;
		box-sizing: content-box;
		outline: 1px solid
			${({ theme, completed }) => (completed ? theme.colors.darkGray : theme.colors.white)};
		background: ${({ theme, completed }) =>
			completed ? theme.colors.darkGray : theme.colors.white};
	}

	:last-child ::after {
		display: none;
	}

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		::after {
			display: none;
		}
	}
`;

const StepLabel = styled.p`
	font-size: 1.4rem;
	color: ${({ theme }) => theme.colors.white};
	font-weight: 700;
	position: absolute;
	bottom: -25px;

	@media (max-width: ${({ theme }) => theme.screens.medium}px) {
		position: relative;
		color: ${({ theme }) => theme.colors.darkGray};
	}
`;

const StepNumber = styled.span`
	background: ${({ theme }) => theme.colors.white};
	display: block;
	text-align: center;
	font-size: 4.5rem;
	height: 6rem;
	width: 6rem;
	line-height: 6rem;
	border-radius: 50%;
	position: relative;

	> svg {
		height: 100%;
	}
`;

const FormWizard = ({ steps, currentStep, onSubmit, onPrev }) => {
	const CurrentFormStep =
		currentStep !== '' ? steps.find((step) => step.slug === currentStep).form : steps[0].form;

	const currentStepSlug = currentStep || steps[0].slug;
	let currentStepIndex = 0;
	steps.forEach((step, i) => {
		if (step.slug === currentStepSlug) {
			currentStepIndex = i;
		}
	});

	const nextStep =
		currentStepIndex === steps.length - 1 ? false : steps[currentStepIndex + 1].slug;
	const prevStep = currentStepIndex === 0 ? false : steps[currentStepIndex - 1].slug;

	const handleSubmit = (data) => {
		window.scrollTo({ top: 0 });
		onSubmit({ data, step: currentStepSlug, nextStep });
	};

	const handlePrev = () => {
		window.scrollTo({ top: 0 });
		onPrev({ step: currentStepSlug, prevStep });
	};

	return (
		<FormWizardContainer>
			<StepsContainer>
				<WebSteps>
					{steps.map((step, i) => {
						const showIcon = i < currentStepIndex || typeof step.icon !== 'undefined';
						const Icon = i < currentStepIndex ? AiOutlineCheck : step.icon || null;
						return (
							<StepItem completed={i <= currentStepIndex} key={step.slug}>
								<div>
									<StepNumber>{showIcon ? <Icon /> : i + 1}</StepNumber>
									<StepLabel>{step.label}</StepLabel>
								</div>
							</StepItem>
						);
					})}
				</WebSteps>
				<MobileSteps>
					<div>
						<StepItem completed>
							<div>
								<StepNumber>{currentStepIndex + 1}</StepNumber>
							</div>
						</StepItem>
						<li>/</li>
						<StepItem>
							<div>
								<StepNumber>{steps.length}</StepNumber>
							</div>
						</StepItem>
					</div>
					<StepLabel>{steps[currentStepIndex].label}</StepLabel>
				</MobileSteps>
			</StepsContainer>

			<Form onSubmit={handleSubmit}>
				{CurrentFormStep && <CurrentFormStep />}
				<Actions center>
					{prevStep && (
						<Button variant="secondary" onClick={handlePrev}>
							Voltar
						</Button>
					)}
					<Button type="submit">Salvar e Continuar</Button>
				</Actions>
			</Form>
		</FormWizardContainer>
	);
};

FormWizard.propTypes = {
	onSubmit: PropTypes.func,
	onPrev: PropTypes.func,
	steps: PropTypes.arrayOf(
		PropTypes.shape({
			label: PropTypes.string.isRequired,
			slug: PropTypes.string.isRequired,
			form: PropTypes.elementType,
			icon: PropTypes.elementType,
		}),
	).isRequired,
	currentStep: PropTypes.string.isRequired,
};

FormWizard.defaultProps = {
	onSubmit: () => {},
	onPrev: () => {},
};

export default FormWizard;
