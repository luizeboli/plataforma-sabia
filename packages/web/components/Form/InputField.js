/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';
import { InputFieldWrapper, InputLabel, InputError, Row } from './styles';
import { validationErrorMessage } from '../../utils/helper';
import Help from './Help';

const StyledInput = styled.input`
	width: 100%;
	height: 4.4rem;
	font-size: 1.4rem;
	margin: 0.5rem 0;
	padding: 1.2rem;
	background: white;
	border: 1px solid ${({ theme }) => theme.colors.mediumGray};
	border-radius: 0.2rem;
	color: ${({ theme }) => theme.colors.lightGray};
`;

const InputField = ({ name, form, type, label, help, validation, ...inputProps }) => {
	const { t } = useTranslation(['error']);
	const { register, errors } = form;

	return (
		<InputFieldWrapper hasError={typeof errors[name] !== 'undefined'}>
			<InputLabel htmlFor={name}>{label}</InputLabel>

			<Row>
				<StyledInput
					id={name}
					type={type}
					name={name}
					aria-label={label}
					aria-required={validation.required}
					ref={register(validation)}
					{...inputProps}
				/>
				{help && <Help id={name} HelpComponent={help} />}
			</Row>

			<InputError>{validationErrorMessage(errors[name], t)}</InputError>
		</InputFieldWrapper>
	);
};

InputField.propTypes = {
	name: PropTypes.string.isRequired,
	label: PropTypes.string,
	placeholder: PropTypes.string,
	icon: PropTypes.func,
	type: PropTypes.string,
	form: PropTypes.shape({
		register: PropTypes.func,
		errors: PropTypes.shape({}),
	}),
	help: PropTypes.node,
	/**
	 * @see https://react-hook-form.com/api#register
	 */
	validation: PropTypes.shape({
		required: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
	}),
};

InputField.defaultProps = {
	form: {},
	type: 'text',
	help: null,
	validation: {},
	label: '',
	placeholder: '',
	icon: () => false,
};

export default InputField;
