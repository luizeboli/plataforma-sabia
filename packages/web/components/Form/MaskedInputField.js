/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import MaskedInput from 'react-input-mask';
import { useTranslation } from 'react-i18next';
import InputField from './InputField';

/**
 * This component works as a wrapper: you can provide all of the props InputField component expects.
 * However, you SHOULD provide mask, pattern and name to the component itself.
 *
 * @param {string} mask Mask to be applied (e.g: 99999-999)
 * @typedef {RegExp} pattern Regex pattern to validate the field when it is submitted
 * @returns {React.Component} An InputField instance.
 */
const MaskedInputField = ({
	mask,
	pattern,
	name,
	alwaysShowMask,
	defaultValue,
	...inputFieldProps
}) => {
	const { t } = useTranslation(['error']);
	const [value, setValue] = useState(defaultValue || '');
	return (
		<MaskedInput
			mask={mask}
			alwaysShowMask={alwaysShowMask}
			onChange={(e) => setValue(e.target.value)}
			value={value}
			name={name}
		>
			{() => (
				<InputField
					name={name}
					{...inputFieldProps}
					validation={{
						...inputFieldProps.validation,
						pattern: {
							value: pattern,
							message: t('invalidPattern'),
						},
					}}
				/>
			)}
		</MaskedInput>
	);
};

MaskedInputField.propTypes = {
	mask: PropTypes.string.isRequired,
	pattern: PropTypes.instanceOf(RegExp).isRequired,
	name: PropTypes.string.isRequired,
	alwaysShowMask: PropTypes.bool,
	defaultValue: PropTypes.string,
};

MaskedInputField.defaultProps = {
	defaultValue: '',
	alwaysShowMask: false,
};

export default MaskedInputField;
