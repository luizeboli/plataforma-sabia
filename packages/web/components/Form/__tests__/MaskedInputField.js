import React from 'react';
import { render, fireEvent, screen } from 'test-utils';
import MaskedInputField from '../MaskedInputField';
import { Form } from '../Form';

const onSubmit = jest.fn();

test('MaskedInputField works as expected if required props are provided', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="cpf"
				mask="999.999.999-99"
				pattern={/^\d{3}\.\d{3}\.\d{3}-\d{2}$/}
			/>
		</Form>,
	);

	expect(container).toMatchSnapshot();
});

test('it shows the mask correctly when no default value is provided and the input is focused', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="cpf"
				placeholder="Your CPF"
				mask="999.999.999-99"
				pattern={/^\d{3}\.\d{3}\.\d{3}-\d{2}$/}
			/>
		</Form>,
	);

	const input = screen.getByPlaceholderText('Your CPF');
	input.focus();
	expect(input.value).toBe('___.___.___-__');

	expect(container).toMatchSnapshot();
});

test('it renders the value correctly if the default value is provided with a Mask', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="cpf"
				defaultValue="123.123.123.32"
				mask="999.999.999-99"
				pattern={/^\d{3}\.\d{3}\.\d{3}-\d{2}$/}
			/>
		</Form>,
	);

	expect(container.querySelector('input[name=cpf]')).toHaveValue('123.123.123-32');
	expect(container).toMatchSnapshot();
});

test('it renders the value correctly if the default value is provided without a Mask', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="cpf"
				defaultValue="12312312332"
				mask="999.999.999-99"
				pattern={/^\d{3}\.\d{3}\.\d{3}-\d{2}$/}
			/>
		</Form>,
	);

	expect(container.querySelector('input[name=cpf]')).toHaveValue('123.123.123-32');
	expect(container).toMatchSnapshot();
});

test('it does not render a value if the typed value is invalid ', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="CEP"
				mask="99999-999"
				pattern={/^\d{5}-\d{3}$/}
				placeholder="Your CEP"
			/>
		</Form>,
	);

	const input = screen.getByPlaceholderText('Your CEP');

	fireEvent.change(input, { target: { value: 'Wrong Value!' } });

	expect(input.value).toBe('_____-___');
	expect(container).toMatchSnapshot();
});

test('it shows validation error on submit if the value is invalid', async () => {
	const { container, findByText } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="CEP"
				mask="99999-999"
				pattern={/^\d{5}-\d{3}$/}
				placeholder="Your CEP"
			/>
		</Form>,
	);

	const input = screen.getByPlaceholderText('Your CEP');
	fireEvent.change(input, { target: { value: '12' } });
	fireEvent.submit(container.querySelector('form'));

	await findByText('invalidPattern');
	expect(onSubmit).not.toHaveBeenCalled();
	expect(container).toMatchSnapshot();
});

test('it renders the Mask correctly as the user types', () => {
	const { container } = render(
		<Form onSubmit={onSubmit}>
			<MaskedInputField
				name="CEP"
				mask="99999-999"
				pattern={/^\d{5}-\d{3}$/}
				placeholder="Your CEP"
			/>
		</Form>,
	);

	const input = screen.getByPlaceholderText('Your CEP');

	fireEvent.change(input, { target: { value: '5' } });
	expect(input.value).toBe('5____-___');

	fireEvent.change(input, { target: { value: '59' } });
	expect(input.value).toBe('59___-___');

	fireEvent.change(input, { target: { value: '59655000' } });
	expect(input.value).toBe('59655-000');

	expect(container).toMatchSnapshot();
});
