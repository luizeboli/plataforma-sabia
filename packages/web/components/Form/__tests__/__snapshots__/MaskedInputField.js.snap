// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaskedInputField works as expected if required props are provided 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="cpf"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="cpf"
          name="cpf"
          placeholder=""
          type="text"
          value=""
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`it does not render a value if the typed value is invalid  1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="CEP"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="CEP"
          name="CEP"
          placeholder="Your CEP"
          type="text"
          value="_____-___"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`it renders the Mask correctly as the user types 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="CEP"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="CEP"
          name="CEP"
          placeholder="Your CEP"
          type="text"
          value="59655-000"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`it renders the value correctly if the default value is provided with a Mask 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="cpf"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="cpf"
          name="cpf"
          placeholder=""
          type="text"
          value="123.123.123-32"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`it renders the value correctly if the default value is provided without a Mask 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="cpf"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="cpf"
          name="cpf"
          placeholder=""
          type="text"
          value="123.123.123-32"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`it shows the mask correctly when no default value is provided and the input is focused 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="cpf"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="cpf"
          name="cpf"
          placeholder="Your CPF"
          type="text"
          value="___.___.___-__"
        />
      </div>
    </div>
  </form>
</div>
`;

exports[`it shows validation error on submit if the value is invalid 1`] = `
.c1 {
  font-size: 1.4rem;
  line-height: 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 > div.wrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 > input {
  border: 1px solid hsla(29,94%,62%);
}

.c1 > .react-select-container .react-select__control {
  border-color: hsla(29,94%,62%);
}

.c2 {
  color: hsla(0,0%,40%);
  font-size: 1.4rem;
  font-weight: 500;
}

.c5 {
  color: hsla(4,85%,57%);
  margin: 0 0 1rem 0;
  display: inline-block;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c4 {
  width: 100%;
  height: 4.4rem;
  font-size: 1.4rem;
  margin: 0.5rem 0;
  padding: 1.2rem;
  background: white;
  border: 1px solid hsla(0,0%,80%);
  border-radius: 0.2rem;
  color: hsla(0,0%,40%);
}

.c4::-webkit-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::-moz-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4:-ms-input-placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c4::placeholder {
  color: hsla(0,0%,52%);
  font-weight: 300;
  font-style: italic;
}

.c0 {
  margin: 1rem auto;
  padding: 2rem 1rem;
  width: 100%;
}

.c0 button,
.c0 a {
  padding-right: 3rem;
  padding-left: 3rem;
}

.c0 > label {
  width: 100%;
  display: block;
  font-weight: 700;
  font-size: 1.7rem;
}

<div>
  <form
    class="c0"
  >
    <div
      class="c1"
    >
      <label
        class="c2"
        for="CEP"
      />
      <div
        class="c3"
      >
        <input
          aria-label=""
          class="c4"
          id="CEP"
          name="CEP"
          placeholder="Your CEP"
          type="text"
          value="12___-___"
        />
      </div>
      <span
        class="c5"
      >
        invalidPattern
      </span>
    </div>
  </form>
</div>
`;
